bundles/net.tourbook.device/src/net/tourbook/device/suunto/Suunto9DeviceDataReader.java

```diff
@@ -0,0 +1,405 @@
+import net.tourbook.importdata.DeviceData;
+import net.tourbook.importdata.SerialParameters;
+import net.tourbook.importdata.TourbookDevice;
+
+public class Suunto9DeviceDataReader extends TourbookDevice {
+
+	private String _importFilePath;
+
+	private final float Kelvin = 273.1499938964845f;
+
+
+	// plugin constructor
+	public Suunto9DeviceDataReader() {
+	}
+
+
+	@Override
+	public String buildFileNameFromRawData(final String rawDataFileName) {
+		// NEXT Auto-generated method stub
+		return null;
+	}
+
+
+	@Override
+	public boolean checkStartSequence(final int byteIndex, final int newByte) {
+		return true;
+	}
+
+
+	public String getDeviceModeName(final int profileId) {
+		return UI.EMPTY_STRING;
+	}
+
+
+	@Override
+	public SerialParameters getPortParameters(final String portName) {
+		return null;
+	}
+
+
+	@Override
+	public int getStartSequenceSize() {
+		return 0;
+	}
+
+
+	public int getTransferDataSize() {
+		return -1;
+	}
+
+
+	@Override
+	public boolean processDeviceData(final String importFilePath, final DeviceData deviceData,
+			final HashMap<Long, TourData> alreadyImportedTours, final HashMap<Long, TourData> newlyImportedTours) {
+
+		if (isValidJSONFile(importFilePath) == false) {
+			return false;
+		}
+		_importFilePath = importFilePath;
+
+		createTourData(alreadyImportedTours, newlyImportedTours);
+		return true;
+	}
+
+
+	@Override
+	public boolean validateRawData(final String fileName) {
+		return isValidJSONFile(fileName);
+	}
+
+
+	/**
+	 * Check if the file is a valid device JSON file.
+	 * 
+	 * @param importFilePath
+	 * @return Returns <code>true</code> when the file contains content with the
+	 *         requested tag.
+	 */
+	protected boolean isValidJSONFile(final String importFilePath) {
+
+		BufferedReader fileReader = null;
+		try {
+
+			String jsonFileContent = GetJsonContentFromGZipFile(importFilePath);
+
+			if (jsonFileContent == null) {
+				return false;
+			}
+
+			try {
+				JSONObject jsonContent = new JSONObject(jsonFileContent);
+				JSONArray samples = (JSONArray) jsonContent.get("Samples");
+
+				String firstSample = samples.get(0).toString();
+				if (firstSample.contains("Lap") && firstSample.contains("Type") && firstSample.contains("Start"))
+					return true;
+
+			} catch (JSONException ex) {
+				return false;
+			}
+
+		} catch (final Exception e) {
+			StatusUtil.log(e);
+		} finally {
+			Util.closeReader(fileReader);
+		}
+
+		return true;
+	}
+
+
+	private void createTourData(final HashMap<Long, TourData> alreadyImportedTours,
+			final HashMap<Long, TourData> newlyImportedTours) {
+
+		String jsonFileContent = GetJsonContentFromGZipFile(_importFilePath);
+
+		// create data object for each tour
+		final TourData tourData = ImportTour(jsonFileContent);
+
+		tourData.setImportFilePath(_importFilePath);
+
+		tourData.setDeviceId(deviceId);
+		// TODO after all data are added, the tour id can be created
+		final Long tourId = tourData.createTourId("49837533398");

```

Comment:
@wolfgang-ch I put a dummy value here because I don't know if that should be a GUID ?
I also see this in Util.Java
public static final String	UNIQUE_ID_SUFFIX_SUUNTO3	= "73198";	//$NON-NLS-1$
Is that Id random ? I was thinking doing this:
public static final String	UNIQUE_ID_SUFFIX_SUUNTO9	= "{RANDOM}";

Reply:
Creating the tour id has historic issues, more can be found in the code https://github.com/wolfgang-ch/mytourbook/blob/5e828ff11281409e332273c85aa91a4f9b8caa12/bundles/net.tourbook/src/net/tourbook/importdata/TourbookDevice.java#L141

You should create the tour id as it is done in the other device importers, e.g. 
https://github.com/wolfgang-ch/mytourbook/blob/5e828ff11281409e332273c85aa91a4f9b8caa12/bundles/net.tourbook.device/src/net/tourbook/device/suunto/Suunto3SAXHandler.java#L471-L472

by creating a unique id in Util. The id is create in Util that duplicates can easily be recognized. (as I can remember this decision after 9 years)

Reply:
Thanks, I will look into that

Reply:
Done